src/NosCore.GameObject/Providers/UpgradeService/UpgradeService.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using ChickenAPI.Packets.ClientPackets.Inventory;
+using ChickenAPI.Packets.Enumerations;
+using ChickenAPI.Packets.Interfaces;
+using ChickenAPI.Packets.ServerPackets.Inventory;
+using ChickenAPI.Packets.ServerPackets.Shop;
+using ChickenAPI.Packets.ServerPackets.UI;
+using NosCore.Core;
+using NosCore.Core.I18N;
+using NosCore.Data.Enumerations.I18N;
+using NosCore.GameObject.ComponentEntities.Extensions;
+using NosCore.GameObject.Helper;
+using NosCore.GameObject.Networking.ClientSession;
+using NosCore.GameObject.Providers.InventoryService;
+using NosCore.GameObject.Providers.ItemProvider.Item;
+using Serilog;
+
+namespace NosCore.GameObject.Providers.UpgradeService
+{
+    public class UpgradeService: IUpgradeService
+    {
+        private readonly ILogger _logger;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unread private field '_logger' or refactor the code to use its value.](https://app.codacy.com/app/NosCoreIO/NosCore/pullRequest?prid=4518428)